pkg/m3admin/client.go

```diff
@@ -167,6 +169,41 @@ func (c *client) DoHTTPRequest(
 	return nil, pkgerrors.WithMessage(ErrNotOk, errMsg)
 }
 
+// DoHTTPJSONPBRequest is a helper for performing a request and
+// parsing the response as a JSONPB message into the response.
+// Both request and response are optional and can be emitted if
+// not wanting to either send or receive message.
+func (c *client) DoHTTPJSONPBRequest(
+	action, url string,
+	request proto.Message,
+	response proto.Message,
```

Comment:
Just curious, is the reason to pass the response so that a user can pass a typed one rather than have to type assert on the way back (if signature returned `(proto.Message, error)`?

Reply:
Well the main thing is I need to unmarshal into a specific proto message, unless the caller wants to do the the whole jsonpb Unmarshal thing (which we're trying to avoid them doing).  That's why they need to provide up front.